### PR TITLE
feat(speculative): add Qwen3 dense target support for EAGLE-1/2/3

### PR DESCRIPTION
**Pull request by khazic**
# What does this PR do?

Add Qwen3 (``Qwen3ForCausalLM``) as a supported target for EAGLE-1 /
EAGLE-2 / EAGLE-3 training. Stacked on top of #2312 (Phi-3 support);
the actual Qwen3-specific delta is one registry entry + three example
configs + docstring updates.

> **Depends on #2312.** Until #2312 lands, the diff shown here also
> includes the Phi-3 PR's commits as a prefix. After #2312 merges this
> branch will rebase onto main and become a 3-line incremental PR.
> The Qwen3-specific commit is `b3d018c8`.

## Changelog (Qwen3 delta only)

- ``components/speculative/eagle/registry.py``: append
  ``Qwen3ForCausalLM`` to ``_DENSE_ARCHITECTURES``. Qwen3 already works
  through the existing config-driven draft path -- it decouples
  ``head_dim`` from ``hidden_size / num_attention_heads``, which the
  attention layer already reads via
  ``getattr(config, "head_dim", ...)``; ``attention_bias`` and
  ``mlp_bias`` are exposed on ``Qwen3Config`` so they are read normally.
- Add example YAMLs:
  ``examples/speculative/eagle{1,2,3}/qwen3_eagle{1,2,3}_perfectblend.yaml``.
- Update draft / recipe docstrings to mention Qwen3 alongside Llama and Phi-3.

No code-path changes were required. The registry dispatch already
exists from #2312.

## Verification

End-to-end smoke test on 8 x H100:

- Target: ``Qwen/Qwen3-8B`` (15.26 GB, 8.19 B params, ``model_type=qwen3``).
- Dataset: PerfectBlend (200-sample slice).
- EAGLE-3 over 25 optimizer steps:

```
2026-05-25 20:34:04 INFO Training start: start_epoch=0 num_epochs=1 batches_per_epoch=25
2026-05-25 20:34:06 INFO epoch=0 step=1  train_loss=9.846308  train_acc=0.000000
2026-05-25 20:34:06 INFO epoch=0 step=2  train_loss=8.477696  train_acc=0.029953
2026-05-25 20:34:07 INFO epoch=0 step=3  train_loss=8.218086  train_acc=0.053741
...
2026-05-25 20:34:13 INFO epoch=0 step=23 train_loss=6.182835  train_acc=0.113525
2026-05-25 20:34:13 INFO epoch=0 step=24 train_loss=6.219086  train_acc=0.098724
2026-05-25 20:34:14 INFO epoch=0 step=25 train_loss=6.175844  train_acc=0.093933
2026-05-25 20:34:14 INFO Epoch 0 done: total_batches_seen=25 global_step=25
```

Loss decreases ``9.85 -> 6.18`` over 25 steps (~37% drop), accuracy ticks
up from ``0`` to ``~0.09``. No ``TypeError`` / ``AttributeError`` at
draft construction, target load (Liger applied to ``model type: qwen3``
without complaint), or training step.

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Contributor guidelines followed
- [ ] Did you write any new necessary tests? No -- end-to-end repro needs
  multi-GPU + a real Qwen3 target; smoke-test evidence above.
- [x] Example configs added under ``examples/speculative/eagle{1,2,3}/``.

### CODE REVIEW

**Review by HuiyingLi — APPROVED** [review]
Thank you so much!
